File: doubt2.cpp
#include<iostream>
using namespace std;

class A{
    public :
    int val1;

    A(int num1){
        val1 = num1;
    }

    A(){

    }

};

class B : public A{
    public :
    int val2;

    B(int num2){
        val2 = num2;
    }

    void printData();
};

void B :: printData(){
    cout<<"The val2 is :"<<val2<<endl;
}

int main(){
    B b(5);
    b.printData();

    
    return 0;
}

//but here always use a blank constructor or use the derived
// constructor method in cpp
//refer doubt2.py

File: doubt2.py
class A :

    val = None

    def __init__(self,val) :
        self.val = val

class B(A) :
    val1 = None
    def __init__(self,val1) :
        self.val1 = val1

    def printData(self) :
        print(self.val1)

b = B(5)

b.printData()
#no need to use any blank constructor if we onlydo the work 
#derived class in pYthon
